### PR TITLE
tls.Dial() validates hostname, no need to do that manually

### PR DESCRIPTION
**Pull request by ash2k**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

`tls.Dial()` validates hostname, no need to do that manually. `VerifyHostname()` is still needed for the `tls.Client()` code path. See https://github.com/kubernetes/kubernetes/pull/109750.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go [hunk 1]
```diff
@@ -101,6 +100,21 @@ func dialURL(ctx context.Context, url *url.URL, transport http.RoundTripper) (ne
 				return nil, err
 			}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go:102]: according to https://github.com/kubernetes/kubernetes/pull/109750, Handshake validates host... why is this still needed for this path?

Comment (liggitt) [reply staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go:102]: also, should this switch to HandshakeContext (and tls.Dial below switch to tls.Dialer#DialContext?)

**Comment by liggitt** [comment]
would like a second set of eyes on this as well
/assign @aojea

**Comment by aojea** [comment]
+1

File: staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go (hunk 1 above)
Comment (ash2k) [reply staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go]: Follow up https://github.com/kubernetes/kubernetes/pull/110079

**Pull request merged**